src/app/dashboard/page.jsx

```diff
@@ -36,7 +36,8 @@ export default function Page() {
   const [userActivity, setUserActivity] = useState([]);
 
   const router = useRouter();
-
+  const params = useSearchParams()
+  const errorSocial = params.get("error_message");
```

Comment:
you should handle the "error" param first, because "error_message" is very Discord specific and it's additional message when there' also "error"